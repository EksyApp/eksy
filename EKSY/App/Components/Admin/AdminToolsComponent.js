import React, {Component} from 'react'
import {Text, View, StyleSheet} from "react-native";
import Header from "../Common/Header";
import * as Theme from "../../Theme";
import Message from '../Common/Message'
import MarkerCardList from '../Common/MarkerCardList'
import Label from "../Common/Label";
import Button from "../Common/Button";
import Input from "../Common/Input";
import Divider from "../Common/Divider";


export default class AdminToolsComponent extends Component {
	

	render () {
		return (
				<View style={styles.container}>
					<Header title='Admin Tools' backButton />
					<Label>Map circle radius</Label>
					<Input
							label='Radius (meters)'
							onChangeText={(radius) => this.props.onRadiusChange(radius)}
							placeholder='Radius'
							keyboardType="numeric"
							defaultValue={this.props.radius + ""}
					/>
					<Divider/>
					<Label>Confirm markers</Label>
					<Button onPress={this.props.onConfirmClick}>
						See unconfirmed markers
					</Button>
				</View>
		)
	}

}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: Theme.backgroundColor,
	},
	
	card: {
		width: '95%',
		alignSelf: 'center'
	}
})
