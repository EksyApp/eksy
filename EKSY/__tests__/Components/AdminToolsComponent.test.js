import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import AdminToolsComponent from "../../App/Components/Admin/AdminToolsComponent";

test('AdminToolsComponent renders correctly', () => {
  const tree = renderer.create(
    <AdminToolsComponent />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
