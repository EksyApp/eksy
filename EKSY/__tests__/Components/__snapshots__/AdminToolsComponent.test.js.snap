// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserMarkersComponent renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(243,244,229)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgb(226,227,217)",
        "height": 50,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            false,
            false,
            false,
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            },
            Object {
              "height": 50,
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": 50,
              "zIndex": 1,
            },
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 35,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "rgb(101,114,122)",
              },
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        î—„
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgb(101,114,122)",
            "fontSize": 20,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Admin Tools
    </Text>
  </View>
  <RCTScrollView
    backButton={true}
    style={
      Array [
        Object {
          "backgroundColor": "rgb(243,244,229)",
          "flex": 1,
        },
        undefined,
      ]
    }
    title="Admin Tools"
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgb(101,114,122)",
              "fontSize": 20,
              "fontWeight": "bold",
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Map circle radius
      </Text>
      <View>
        <View
          style={
            Array [
              Object {},
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "#86939e",
                    "fontSize": 15,
                    "fontWeight": "bold",
                    "marginBottom": 1,
                    "marginLeft": 20,
                    "marginRight": 20,
                    "marginTop": 15,
                  },
                  Object {
                    "color": "rgb(101,114,122)",
                    "fontSize": 20,
                  },
                  undefined,
                ],
              ]
            }
          >
            Radius (meters)
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "borderBottomColor": "#bdc6cf",
                "borderBottomWidth": 1,
                "marginLeft": 20,
                "marginRight": 20,
              },
              Object {
                "backgroundColor": "rgb(226,227,217)",
                "margin": 10,
              },
            ]
          }
        >
          <TextInput
            defaultValue="undefined"
            keyboardType="numeric"
            label="Radius (meters)"
            onChangeText={[Function]}
            placeholder="Radius"
            selectionColor="#86939e"
            style={
              Array [
                Object {
                  "color": "#86939e",
                  "fontSize": 17.5,
                  "height": 36,
                  "width": 750,
                },
                Object {
                  "color": "rgb(89,85,78)",
                  "fontSize": 18,
                },
              ]
            }
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#e1e8ee",
              "height": 1,
            },
            Array [
              Object {
                "backgroundColor": "rgb(112,112,112)",
                "height": 1,
                "margin": 3,
                "marginLeft": 10,
                "marginRight": 10,
              },
              undefined,
            ],
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgb(101,114,122)",
              "fontSize": 20,
              "fontWeight": "bold",
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Confirm markers
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "rgb(230,232,227)",
            "borderColor": "rgb(101,114,122)",
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 40,
            "marginBottom": 15,
            "marginLeft": 5,
            "marginRight": 5,
            "marginTop": 5,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "rgb(101,114,122)",
              "fontSize": 16,
              "fontWeight": "600",
              "paddingBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          See unconfirmed markers
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
